ibm/service/cos/data_source_ibm_cos_bucket.go

```diff
@@ -556,6 +556,100 @@ func DataSourceIBMCosBucket() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"lifecycle_rule": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"abort_incomplete_multipart_upload": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"days_after_initiation": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"expiration": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"date": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"days": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"expired_object_delete_marker": {
+										Type:     schema.TypeBool,
+										Computed: true, // API returns false; conflicts with date and days
+									},
+								},
+							},
+						},
+						"filter": {
+							Type:     schema.TypeList,
+							Computed: true,
+							// IBM has filter a required parameter
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"prefix": {
+										Type:     schema.TypeString, // check if prefix empty is eccepted and if filter empty is accepted
+										Computed: true,
+									},
+								},
+							},
+						},
+						"rule_id": {
+							Type:     schema.TypeString,
+							Computed: true, // check with team if we have any validation function
```

Comment:
Remove the comment and did we decide on the validation part?

Reply:
this is the data source so validation is not required , I have added the validation in the resource file.